bioptim/interfaces/SolverOptions.py

```diff
@@ -0,0 +1,95 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+
+
+@dataclass
+class SolverOptions(ABC):
+    @abstractmethod
+    def set_convergence_tol(self, tol):
+        """
+         This function set the convergence tolerance
+        """
+
+    @abstractmethod
+    def set_constraint_tol(self, tol):
+        """
+         This function set the constraint tolerance.
+        """
+
+    @abstractmethod
+    def set_solver(self, name):
+        """
+         This function set the solver
+        """
+
+    def set_max_iter(self, num):
+        """
+         This function set the maximal number of iterations
+        """
+
+
+@dataclass
+class SolverOptionsIPOPT(SolverOptions):
```

Comment:
CamelCase usually requires acronyms and abbreviations to be uncapitalized (so SolveOptionsIpopt)